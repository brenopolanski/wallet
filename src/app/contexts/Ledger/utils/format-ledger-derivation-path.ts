import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
