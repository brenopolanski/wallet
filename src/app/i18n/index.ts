import i18n from "i18next";
import { initReactI18next } from "react-i18next";

// Helpers
import { buildTranslations } from "./helpers";

const defaultNS = "translation";

const resources = {
	en: {
		translation: buildTranslations(),
	},
};

// eslint-disable-next-line
i18n.use(initReactI18next).init({
	defaultNS,
	lng: "en",
	ns: [defaultNS],
	resources,
});

export { defaultNS, i18n, resources };
