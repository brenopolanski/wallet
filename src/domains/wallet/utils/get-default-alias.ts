import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
