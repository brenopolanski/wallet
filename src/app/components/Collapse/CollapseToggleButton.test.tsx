import { translations } from "app/i18n/common/i18n";
import React from "react";
import { fireEvent, render, screen } from "utils/testing-library";

import { CollapseToggleButton } from "./CollapseToggleButton";

describe("CollapseToggleButton", () => {
	it("should render", () => {
		const onClick = jest.fn();

		render(<CollapseToggleButton isOpen={false} onClick={onClick} />);

		const button = screen.getByTestId("CollapseToggleButton");

		expect(button).toHaveTextContent(translations.SHOW);

		fireEvent.click(button);

		expect(onClick).toHaveBeenCalledWith(expect.objectContaining({ nativeEvent: expect.any(MouseEvent) }));
	});

	it("should render open", () => {
		render(<CollapseToggleButton isOpen={true} />);

		const button = screen.getByTestId("CollapseToggleButton");

		expect(button).toHaveTextContent(translations.HIDE);
	});

	it("should render disabled", () => {
		const { container } = render(<CollapseToggleButton isOpen={true} disabled />);

		expect(container).toMatchSnapshot();
	});
});
