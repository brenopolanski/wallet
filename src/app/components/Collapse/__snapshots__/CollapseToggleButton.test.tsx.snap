// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapseToggleButton should render disabled 1`] = `
<div>
  <button
    class="flex items-center py-2 font-semibold rounded focus:outline-none space-x-2 text-theme-secondary-500"
    data-testid="CollapseToggleButton"
    disabled=""
  >
    <span>
      Hide
    </span>
    <span
      class="CollapseToggleButton__ToggleIcon-sc-1wkp6k4-0 hoUNik"
    >
      <div
        class="sc-bdvvtL Cfvxc p-1"
        height="10"
        width="10"
      >
        <svg>
          chevron-down-small.svg
        </svg>
      </div>
    </span>
  </button>
</div>
`;
